### ISSUE TITLE
Port Forwarding Seems Broken

### ISSUE THREAD

**Issue by leonjza** [ContainerSSH/ContainerSSH#527]
I know it is not officially released yet, but it looks like SSH port forwarding is not working, even with all of the forwarding options enabled in the `security.forwarding` configuration.

## To Reproduce

On macOS running Docker desktop, I compiled containerssh from source (at [279bb9a461166](https://github.com/ContainerSSH/ContainerSSH/tree/279bb9a4611667f193884c3dad971568300c053f)) (using `go build ./cmd/containerssh`) which pins `libcontainerssh` to [e8a303b04b01](https://github.com/ContainerSSH/libcontainerssh/tree/e8a303b04b017a7a25f37f84c6d349442a08d9e3) which includes https://github.com/ContainerSSH/libcontainerssh/pull/103 to forward a port inside the container to the client.

I am using the following configuration file:

```yaml
---
ssh:
  banner: "Welcome to ContainerSSH!\n"
  hostkeys:
    - conf/hostkey
security:
  forwarding:
    reverseForwardingMode: enable
    forwardingMode: enable
    socketForwardingMode: enable
    socketListenMode: enable
    x11ForwardingMode: enable
log:
  level: debug
auth:
  url: "http://localhost:8080"
configserver:
  url: "http://localhost:8080/config"
backend: docker
docker:
  connection:
    host: unix:///var/run/docker.sock
```

I am starting containerssh with: `./containerssh --config conf/config.yaml`, using the example authentication and configuration server from [here](https://github.com/ContainerSSH/ContainerSSH/tree/main/cmd/containerssh-testauthconfigserver).

Once started, I would SSH in with `ssh -L 1111:localhost:1111 foo@localhost -p 2222`. I would then install `ncat` with `apt update && apt install -y ncat` inside of the container and then start ncat with `ncat -lvp 1111`.

On my host, I would then try and connect to the forwarded port with `ncat localhost 1111`. This would fail with an SSH client error in the same console: `channel 4: open failed: connect failed: Error setting up the forwarding`.

The containerssh debug log has the following:

```json
{"timestamp":"2022-09-18T20:59:20+02:00","level":"debug","code":"DOCKER_EXEC","message":"Creating and attaching to container exec...","details":{"authenticatedUsername":"foo","backend":"docker","connectionId":"35deb8c7705c4097959bfc2cc564cacf","containerId":"d31c55d3036d6d0326165bf6c53ff1060c625e88348ffcadc8e701522f5e8676","module":"backend","remoteAddr":"::1","username":"foo"}}
{"timestamp":"2022-09-18T20:59:20+02:00","level":"debug","code":"DOCKER_EXEC_CREATE","message":"Creating exec...","details":{"authenticatedUsername":"foo","backend":"docker","connectionId":"35deb8c7705c4097959bfc2cc564cacf","containerId":"d31c55d3036d6d0326165bf6c53ff1060c625e88348ffcadc8e701522f5e8676","module":"backend","remoteAddr":"::1","username":"foo"}}
{"timestamp":"2022-09-18T20:59:20+02:00","level":"debug","code":"DOCKER_EXEC_ATTACH","message":"Attaching exec...","details":{"authenticatedUsername":"foo","backend":"docker","connectionId":"35deb8c7705c4097959bfc2cc564cacf","containerId":"d31c55d3036d6d0326165bf6c53ff1060c625e88348ffcadc8e701522f5e8676","module":"backend","remoteAddr":"::1","username":"foo"}}
{"timestamp":"2022-09-18T20:59:20+02:00","level":"info","code":"DOCKER_AGENT_LOG","message":"Error: invalid mode: forward-server\n","details":{"authenticatedUsername":"foo","backend":"docker","connectionId":"35deb8c7705c4097959bfc2cc564cacf","module":"backend","remoteAddr":"::1","username":"foo"}}
{"timestamp":"2022-09-18T20:59:20+02:00","level":"debug","code":"SSH_NEW_CHANNEL_REJECTED","message":"New forwarding channel rejected (Error setting up the forwarding)","details":{"channelId":1,"connectionId":"35deb8c7705c4097959bfc2cc564cacf","module":"ssh","remoteAddr":"::1","type":"direct-tcpip","username":"foo"}}
```

Screenshot of the `ncat` server / client at the time of the above log entry.

<img width="1021" alt="image" src="https://user-images.githubusercontent.com/1148127/190924147-e05b3295-1390-4916-97d4-f0ee6374c44d.png">

## Expected behavior

The `ncat` client to connect to the `ncat` server running inside of the container.

## Version

- containerssh [279bb9a461166](https://github.com/ContainerSSH/ContainerSSH/tree/279bb9a4611667f193884c3dad971568300c053f)
- libcontainerssh [e8a303b04b01](https://github.com/ContainerSSH/libcontainerssh/tree/e8a303b04b017a7a25f37f84c6d349442a08d9e3)

**Comment by fschaefer** [comment]
Thank you @janosdebugs for the helping hand! Yes, I'm sorry. I just forgot about Slack while browsing GitHub on how to build the project. Maybe something to consider for the README. Anyway, your comment will help me a lot. Many thanks!

**Comment by tsipinakis** [comment]
The forwarding feature has now been released with 0.5. No need for a dev environment anymore to have access to it :) (Unless you want to contribute!)
